include/zephyr/arch/x86/x86_acpi_osal.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/arch/x86/efi.h>
+#include <zephyr/arch/x86/legacy_bios.h>
+
+#ifndef ZEPHYR_ARCH_X86_INCLUDE_X86_ACPI_H_
+#define ZEPHYR_ARCH_X86_INCLUDE_X86_ACPI_H_
+
+uint32_t bios_acpi_rsdp_get(void);
+
+#if defined(CONFIG_X86_EFI)
```

Comment:
No you can't assume that once EFI is enabled, the system will always get it successfully from EFI.

Look at the original code you removed in arch/x86/core/acpic.c: is efi fails to get rsdp, it falls back to legacy lookup.